src/org/openlcb/can/CanInterface.java

```diff
@@ -35,7 +39,17 @@
 
     boolean initialized = false;
 
-    public CanInterface(NodeID interfaceId, CanFrameListener frameOutput) {
+    private static ThreadPoolExecutor threadPool = null;
+    final static int minThreads = 10;
+    final static int maxThreads = 10;
+    final static long threadTimeout = 10; // allowed idle time for threads, in seconds.
+
+    public CanInterface(NodeID interfaceId, CanFrameListener frameOutput){
+          this(interfaceId,frameOutput, new ThreadPoolExecutor(minThreads,maxThreads,threadTimeout,TimeUnit.SECONDS,new LinkedBlockingQueue<Runnable>()));
```

Comment:
The intention here is that the two parameter version of the constructor will be deprecated in favor of the three parameter version.  I'll add the deprecation annotations.

The idea is we only want one threadpool per connection, so the name will come from the OlcbInterface.

Reply:
That sounds good. We create the olcbinterface though so it seems more likely that we want to pass in the threadpool to it?

Reply:
In the current head of master, the OlcbInterface creates both the CanInterface and the DatagramMeteringBuffer, so that is where I created the ThreadPoolExecutor.

I actually think there is a design problem here though, which I am not correcting now.  Because the OlcbInterface creates the CanInterface, OlcbInterface is very CAN centric.  CanInterface really should be derived from OlcbInterface.

Reply:
Where do you see OlcbInterface creating a CanInterface?
I see the opposite, CanInterface creates the (more generic) OlcbInterface (https://github.com/openlcb/OpenLCB_Java/blob/master/src/org/openlcb/can/CanInterface.java#L44)

OlcbInterface does not have any imports from the can.* package and I see nothing CAN-specific in there.

The way JMRI uses this library is a different story, in JMRI OpenLCB system connection derives from Can System Connection and everything there is very CAN-specific. But that's a very different design problem.

Reply:
I need to look at that again.  It has been over a week since I was really looking at this code.

But this is where the current design leads to confusion.

I see the CanInterface as having an “is a” relationship with the OlcbInterface, but the code you pointed to shows it having a “has a” relationship.